tests/gtest/tcp/tcp_sockopt.cc

```diff
@@ -0,0 +1,324 @@
+                    std::make_tuple(AF_INET, IPPROTO_TCP, TCP_KEEPCNT,
+                                    std::numeric_limits<int8_t>::max() + 1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPIDLE, -1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPIDLE,
+                                    std::numeric_limits<int16_t>::max() + 1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPINTVL, -1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPINTVL, 0),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPINTVL,
+                                    std::numeric_limits<int16_t>::max() + 1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPCNT, -1),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPCNT, 0),
+                    std::make_tuple(AF_INET6, IPPROTO_TCP, TCP_KEEPCNT,
+                                    std::numeric_limits<int8_t>::max() + 1)));
+
+using getscokopt_params = std::tuple<int, int, int, const char *>;
+using tcp_getsockopt_equal_to_file = testing::TestWithParam<getscokopt_params>;
+TEST_P(tcp_getsockopt_equal_to_file, default_value) {
+  int socket_domain, level, optname;
+  const char *file_path;
+  std::tie(socket_domain, level, optname, file_path) = GetParam();
+
+  auto fd = reusable_cleanable_test_socket(socket_domain, SOCK_STREAM, 0);
+  EXPECT_GE(fd, 0) << "socket syscall failed to setup a socket";
+
+  /* Get the value via getsockopt */
+  int getsockopt_value = -1;
+  socklen_t actual_enable_len = sizeof(getsockopt_value);
+  auto result =
+      getsockopt(fd, level, optname, &getsockopt_value, &actual_enable_len);
+  EXPECT_EQ(result, 0) << "getsockopt failed";
+  EXPECT_EQ(actual_enable_len, sizeof(getsockopt_value))
+      << "Got unexpected size of agument";
+
+  /* Get the value from the file */
+  int file_value = -1;
+  EXPECT_TRUE(bool(std::ifstream{file_path} >> file_value))
+      << "Failed reading the file";
+
+  ASSERT_EQ(getsockopt_value, file_value)
+      << "The values in the file and the getsockopt differ";
+  close(fd);
+}
+INSTANTIATE_TEST_CASE_P(
+    keep_alive, tcp_getsockopt_equal_to_file,
+    testing::Values(
+        std::make_tuple(AF_INET, IPPROTO_TCP, TCP_KEEPIDLE,
+                        "/proc/sys/net/ipv4/tcp_keepalive_time"),
+        std::make_tuple(AF_INET, IPPROTO_TCP, TCP_KEEPINTVL,
+                        "/proc/sys/net/ipv4/tcp_keepalive_intvl"),
+        std::make_tuple(AF_INET, IPPROTO_TCP, TCP_KEEPCNT,
+                        "/proc/sys/net/ipv4/tcp_keepalive_probes")));
+
+using setsockopt_param = std::tuple<int, int, int>;
+class tcp_with_fifo : public testing::TestWithParam<setsockopt_param> {
+protected:
+  ipc m_ipc_server_to_client{};
+
+  void SetUp() override { m_ipc_server_to_client.create(); }
+
+  void TearDown() override { m_ipc_server_to_client.reset(); }
+};
+
+TEST_P(tcp_with_fifo, set_listen_get_accept_socket) {
```

Comment:
Please consider adding test description as in referenced https://github.com/Mellanox/libvma/blob/master/tests/gtest/tcp/tcp_connect.cc#L138